src/opentype/woff2.js

```diff
@@ -68,38 +88,44 @@ class Woff2TableDirectoryEntry {
       this.tag = getWOFF2Tag(tagNumber);
     }
 
+    /*
+        "Bits 6 and 7 indicate the preprocessing transformation version number (0-3)
+        that was applied to each table. For all tables in a font, except for 'glyf'
+        and 'loca' tables, transformation version 0 indicates the null transform
+        where the original table data is passed directly to the Brotli compressor
+        for inclusion in the compressed data stream. For 'glyf' and 'loca' tables,
+        transformation version 3 indicates the null transform"
+    */
+    const transformVersion = (this.transformVersion = (this.flags & 192) >> 6);
+    let hasTransforms = transformVersion !== 0;
+    if (this.tag === `glyf` || this.tag === `loca`) {
+      hasTransforms = this.transformVersion !== 3;
+    }
+
     this.origLength = p.uint128;
-    const pptVersion = (this.pptVersion = this.flags >> 6);
-    if (
-      pptVersion !== 0 ||
-      ((this.tag === "glyf" || this.tag === "loca") && pptVersion !== 3)
-    ) {
+    if (hasTransforms) {
       this.transformLength = p.uint128;
     }
-    this.length = p.offset; // FIXME: we can probably calculat this without asking the parser
   }
 }
 
 /**
  * Build late-evaluating properties for each table in a
- * woff/woff2 font, so that accessing a table via the
- * woff.tables.tableName or woff2.tables.tableName
- * property kicks off a table parse on first access.
+ * woff2 font, so that accessing a table via the
+ * font.opentype.tables.tableName property kicks off
+ * a table parse on first access.
  *
- * @param {*} woff the woff or woff2 font object
- * @param {DataView} dataview passed when dealing with woff
- * @param {buffer} decoded passed when dealing with woff2
+ * @param {*} woff2 the woff2 font object
+ * @param {decoded} the original (decompressed) SFNT data
+ * @param {createTable} the opentype table builder function
  */
 function buildWoff2LazyLookups(woff2, decoded, createTable) {
   woff2.tables = {};
   woff2.directory.forEach((entry) => {
     lazy(woff2.tables, entry.tag.trim(), () => {
-      const useTransform = typeof entry.transformLength !== "undefined";
-      const data = decoded.slice(
-        entry.origOffset,
-        entry.origOffset +
-          (useTransform ? entry.transformLength : entry.origLength)
-      );
+      const start = entry.offset;
+      const end = start + (entry.transformLength ? entry.transformLength : entry.origLength);
+      const data = decoded.slice(start, end);
```

Comment:
Note that this does not have any code in place to deal with transformed `glyf`, `loca`, or `hmtx` tables yet. I've filed https://github.com/Pomax/lib-font/issues/118 to address that in the future.